build.sh

```diff
@@ -285,7 +285,7 @@ build_native()
             source "$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh" "$__ProjectRoot" $__ClangMajorVersion "$__ClangMinorVersion" $platformArch $__BuildType $__CodeCoverage $scan_build $generator "$extraCmakeArguments" "$__cmakeargs"
         else
             echo "Invoking \"$__ProjectRoot/src/pal/tools/gen-buildsys-gcc.sh\" \"$__ProjectRoot\" $__GccMajorVersion \"$__GccMinorVersion\" $platformArch $__BuildType $__CodeCoverage $generator $extraCmakeArguments $__cmakeargs"
-            source "$__ProjectRoot/src/pal/tools/gen-buildsys-gcc.sh" "$__ProjectRoot" "$__GccMajorVersion" "$__CGccMinorVersion" $platformArch $__BuildType $__CodeCoverage $generator "$extraCmakeArguments" "$__cmakeargs"
+            "$__ProjectRoot/src/pal/tools/gen-buildsys-gcc.sh" "$__ProjectRoot" "$__GccMajorVersion" "$__CGccMinorVersion" $platformArch $__BuildType $__CodeCoverage $generator "$extraCmakeArguments" "$__cmakeargs"
```

Comment:
Thanks, I was able to reproduce it. Maybe we can remove `source` from line 285 as well, as the intention is most probably to call the script with arguments, as opposed to source it to call its functions later?

Reply:
@janvorli : what do you think about line 285?

Reply:
I've just added the `source` in a change I've merged in today. I need it for the clang script so that it can export variables SCAN_BUILD_COMMAND, CCC_CC and CCC_CXX that I need to use for the clang static analyzer invocation. I used source for gcc script too for symmetry.

Reply:
ok, let's not touch clang but revert the source in GCC as it breaks the build?

Reply:
But doesn't it break it just because the script path detection doesn't work in case we use the source? Should we fix that instead so that it works in that mode too? Sourcing it for one compiler and not doing that for another looks strange.

Reply:
@am11 do you have a fix that works for non-BASH?

Reply:
Full description is here: https://stackoverflow.com/a/29835459
I think options with POSIX are:

* drop the `source`.
* with `source` or `.`, pass the target script directory as argument from callsite
* with `source` adapt `rreadlink()` script from SO answer

Reply:
what's your preference @janvorli ?

Reply:
I'm leaning towards #2